### DISCUSSION TITLE
Why isn't there a complex format in NetCDF?

### DISCUSSION THREAD

**Discussion by pabloconrat** [Unidata/netcdf-c#2434] [Ideas]
Hi, maybe this is naive, but I feel like a lot of people use complex data and the NetCDF data format. They are all programming work-arounds which seems quite inefficient and error-prone. Is there a reason why there isn't a native format?

**Comment by edwardhartnett** [comment]
Interesting suggestion!

Just to flesh this out, what would such a format be? Two NC_DOUBLES? Or would the user need to be able to select between DOUBLE and FLOAT, so we really would want two types:

NC_COMPLEX_FLOAT
NC_COMPLEX_DOUBLE

Such a type can be constructed as a user-defined type with netCDF-4 and used pretty easily. Have you tried that? I would suggest you look at the nc_def_var_compound() function.
